src/main/java/in/projecteka/consentmanager/consent/ConsentRequestController.java

```diff
@@ -0,0 +1,37 @@
+package in.projecteka.consentmanager.consent;
+
+import in.projecteka.consentmanager.consent.model.request.ConsentRequest;
+import in.projecteka.consentmanager.consent.model.response.ConsentRequestResponse;
+import in.projecteka.consentmanager.consent.repository.ConsentRequestRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Mono;
+
+import javax.validation.Valid;
+import java.util.UUID;
+
+@RestController
+@Validated
+public class ConsentRequestController {
+
+    @Autowired
```

Comment:
Instead of auto wiring, can we create a bean?

Reply:
the repository is created through the java config. if we want constructor injection onto controller, then we need to move that to the java config (ConsentManagerConfiguration) as well. The servcie impl will be created through java config (not done yet)